lib/upload.js

```diff
@@ -749,11 +786,25 @@ class BaseUpload {
           req.setHeader('Upload-Length', this._size)
         }
 
-        if (value === null) {
+        if (this.options.enableChecksum) {
+          if (this._checksum) {
+            value.arrayBuffer().then((chunkBuffer) => {
```

Comment:
Could this entire if block be merged with the last else block? They seem very similar and would avoid some duplication.

Reply:
Not sure how this might reduce the duplication as one is async. I've removed one outer if condition. Is that okay?